internal/monitors/conviva/conviva.go

```diff
@@ -0,0 +1,254 @@
+				}
+			}
+			for i := range dps {
+				m.Output.SendDatapoint(dps[i])
+			}
+		}(m, conf, url)
+	}
+}
+
+func getSendMetriclensMetrics(maxNumOfGoroutinesChan chan struct{}, m *Monitor, metricConfig *MetricConfig, conf *Config)  {
+	ctx, cancel := context.WithTimeout(m.ctx, time.Duration(conf.IntervalSeconds) * time.Second)
+	for _, metriclensDimension := range metricConfig.Dimensions {
+		select {
+		case <- ctx.Done():
+			cancel()
+			logger.Error(ctx.Err())
+			return
+		case maxNumOfGoroutinesChan <- struct{}{}:
+			url := "https://api.conviva.com/insights/2.4/metrics.json?metrics=" + metricConfig.Metric +
+				"&filter_ids=" + strings.Join(metricConfig.metriclensFilterIds, ",") +
+				"&metriclens_dimension_id=" + conf.dimensionIdByAccountAndName[metricConfig.Account][metriclensDimension]
+			go func(m *Monitor, conf *Config, url string, metricConfig *MetricConfig, metriclensDimension string) {
+				defer func() {<- maxNumOfGoroutinesChan}()
+				res, err := get(m, conf, url)
+				if err != nil {
+					logger.Error(err)
+					return
+				}
+				tablesByFilterId := res[metricConfig.Metric].(map[string]interface{})["tables"].(map[string]interface{})
+				metriclensDimensionEntities := res[metricConfig.Metric].(map[string]interface{})["xvalues"].([]interface{})
+				dps := make([]*datapoint.Datapoint, 0)
+				for filterId, table := range tablesByFilterId {
+					for tableKey, tableValue := range table.(map[string]interface{}) {
+						if tableKey == "rows" {
+							for rowIndex, row := range tableValue.([]interface{}) {
+								select {
+								case <-ctx.Done():
+									cancel()
+									logger.Error(ctx.Err())
+									return
+								default:
+									if row != nil {
+										for metricIndex, metricValue := range row.([]interface{}) {
+											dps = append(dps, sfxclient.GaugeF(
+												conf.metriclensMetricNames[metricConfig.Metric][metricIndex],
+												map[string]string{
+													"metric":  metricConfig.Metric,
+													"account": metricConfig.Account,
+													"filter":  conf.filterNameById[metricConfig.Account][filterId],
+													metriclensDimension: metriclensDimensionEntities[rowIndex].(string),
+												},
+												metricValue.(float64)))
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+				now := time.Now()
+				for i := range dps {
+					dps[i].Timestamp = now
+					dps[i].Meta[dpmeta.NotHostSpecificMeta] = true
+					m.Output.SendDatapoint(dps[i])
+				}
+			}(m, conf, url, metricConfig, metriclensDimension)
+		}
+	}
+}
+
+func get(m *Monitor, conf *Config, url string) (map[string]interface{}, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(conf.Username, conf.Password)
+	res, err := m.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	var payload map[string]interface{}
```

Comment:
It is much better to pass in an object of a struct type that conforms to the json structure you expect instead of deserializing to `interface{}` objects and doing type assertions.

See the way the kubelet monitor does it in `internal/monitors/cadvisor/kubeletstats.go:158`